### PR TITLE
Performance: Add isLazy property to Tabs

### LINKED ISSUE #1384: [Tabs] Tabs get mounted when they are inactive
Hi there!
First of all I would like to thank you for this fantastic library. It's really impressive!

**Expected Behavior**

The Tabs component only mounts the contents of a `TabPanel` to the DOM when the tab is activated/clicked/selected.

**Link to minimal reproduction**
https://codesandbox.io/s/chakra-tabs-mounting-early-kcign

**Steps to Reproduce**
1. Open the [codesandbox](https://codesandbox.io/s/chakra-tabs-mounting-early-kcign)
2. Open the console view
3. See console logs from both tabs, although we only ever activated the first one

**Suggested solution(s)**

Instead of using `display: none` to hide the DOM elements, there should also be an option to not mount the DOM elements at all until the tab is activated.

**Desktop (please complete the following information):**

- OS: MacOS 10.15.5
- Browser: Chromium
- Version: 84.0.4132.0

**Additional context**

I'm currently using the Tabs component in an application where every tab fetches *a lot* of data. The fact that the Tabs component renders all tabs to the DOM and thereby fires lots of HTTP requests simultaneously is causing performance issues.
I have tried to use the `isManual` option, but the tabs still get rendered to the DOM.

### PR DESCRIPTION
**Pull request by hericlesme**
### Description
Add `isLazy` property to `Tabs` to only mount the contents of a TabPanel to the DOM when the tab is activated/clicked/selected.

### Linked Issues
Fixes #1384

### How has this been tested?
A new test has been added in `packages/tabs/tests/tabs.test.tsx`

#
Related:  #1118

### CODE REVIEW

File: packages/tabs/src/use-tabs.ts [hunk 1]
```diff
@@ -57,6 +57,12 @@ export interface UseTabsProps {
    * The id of the tab
    */
   id?: string
+  /**
+   * Performance 🚀:
+   * If `true`, the MenuItem rendering will be deferred
+   * until the menu is open.
```
Comment (with-heart) [inline packages/tabs/src/use-tabs.ts:63]: Should probably refer to these as tabs :)

**Comment by jmiazga** [comment]
I recently used something similar to this in react-bootstrap, but it was broken up as 2 props, `mountOnEnter` and `unmountOnExit`. I wonder if we want to do something like that?

https://react-bootstrap.github.io/components/tabs/#tabs-props

**Comment by hericlesme** [comment]
> Very nice @hericlesme! Could you add a note about this to the docs as well?

Sure thing, @with-heart!

File: packages/tabs/src/use-tabs.ts (hunk 1 above)
Comment (hericlesme) [reply packages/tabs/src/use-tabs.ts]: Oops... Done!

**Comment by with-heart** [comment]
@jmiazga imo that's probably overkill!

**Comment by jmiazga** [comment]
@with-heart That's fair. The only reason I call that out is if a component has a `useEffect` that makes an api call or something, that call will fire every time the component is mounted. That will force the user to pull that call up from the tab.

**Comment by with-heart** [comment]
@jmiazga You raise a good point! Hadn't thought about that. I think it's still fine though, because it's pretty easy to lift something like that up into a callback passed to the component rendered by the tab.

I feel like we should go with this for now and if it turns out to be a common frustration to have to do something like that, we can add the other two later.

**Comment by hericlesme** [comment]
Docs added, @with-heart :book: :tada:

File: website/pages/docs/components/tabs.mdx [hunk 2]
```diff
@@ -310,6 +310,31 @@ tab by pressing <kbd>Space</kbd> or <kbd>Enter</kbd>.
 </Tabs>
 ```
 
+### Lazily mounting tab panels
+
+By default, the Tabs component renders all tabs to the DOM If you'd like to only mount the contents of a `TabPanel` to the DOM at once.
+If you want to defer rendering of content until the tab is activated/clicked/selected, you can use the `isLazy` property.
+In this way, the tab content will not be mounted to the DOM until it is open.
```
Comment (with-heart) [inline website/pages/docs/components/tabs.mdx:317]:
```suggestion
By default, the `Tabs` component renders all tabs content to the DOM, meaning that invisible tabs are still rendered but are hidden by styles.

If you want to defer rendering of each tab until that tab is selected, you can use the `isLazy` prop. This is useful if your tabs require heavy performance, or make network calls on mount that should only happen when the component is displayed.
```

Trying to edit this a little bit to help explain why you'd want to use it. How does this sound @ljosberinn?

Comment (ljosberinn) [reply website/pages/docs/components/tabs.mdx:317]: lgtm!

**Pull request merged**